File: apps/macos/PasteSheets/Domain/Repositories/SettingsRepository.swift
import Foundation

protocol SettingsRepository {
    func getSetting(key: String) throws -> String?
    func setSetting(key: String, value: String) throws
}

final class SettingsRepositoryImpl: SettingsRepository {
    private let dataSource: SettingsDataSource

    init(dataSource: SettingsDataSource = SettingsDataSourceImpl()) {
        self.dataSource = dataSource
    }

    func getSetting(key: String) throws -> String? {
        try dataSource.get(key: key)
    }

    func setSetting(key: String, value: String) throws {
        try dataSource.set(key: key, value: value)
    }
}

File: apps/windows/PasteSheet/Domain/Repositories/SettingsRepository.cs
using PasteSheet.Data.DataSources;

namespace PasteSheet.Domain.Repositories;

public interface ISettingsRepository
{
    string? GetSetting(string key);
    void SetSetting(string key, string value);
}

public sealed class SettingsRepository : ISettingsRepository
{
    private readonly ISettingsDataSource _dataSource;

    public SettingsRepository(ISettingsDataSource dataSource) => _dataSource = dataSource;

    public string? GetSetting(string key) => _dataSource.Get(key);

    public void SetSetting(string key, string value) => _dataSource.Set(key, value);
}
